src/platform/silabs/wifi/SiWx/WifiInterfaceImpl.cpp

```diff
@@ -327,14 +327,17 @@ sl_status_t ScanCallback(sl_wifi_event_t event, sl_wifi_scan_result_t * scan_res
     sl_status_t status = SL_STATUS_OK;
     if (SL_WIFI_CHECK_IF_EVENT_FAILED(event))
     {
-        ChipLogError(DeviceLayer, "Scan Netwrok Failed: 0x%lx", *reinterpret_cast<sl_status_t *>(status));
+        if (scan_result != nullptr)
+        {
+            status = *reinterpret_cast<sl_status_t *>(scan_result);
```

Comment:
what is happening here?
casting scan_result as a sl_status_t pointer to then use its value at address to then cast it as a uint32 in the next step.

Reply:
Removed the reinterpret casting.

In case of scan event failures, scan_result will be of type sl_status_t, so it is typecast to sl_status_t

`  In case of event failure, SL_WIFI_FAIL_EVENT_STATUS_INDICATION bit is set in the event.
 When this bit is set, the `data` parameter will be of type `sl_status_t`, and the `data_length` parameter can be ignored.`